app.js

```diff
@@ -38,6 +54,118 @@ app.post('/vegetables', helpers.validate, (req, res, next) => {
   res.status(201).json(vegetable)
 })
 
+// delete a vegetable
+app.delete('/vegetables/:id', (req, res, next) => {
+  const { vegetables } = data
+  const { id } = req.params
+
+  // if the vegetable exists, delete from the array, return the vegetable
+  if (vegetables.find(veggie => veggie.id === id)) {
+    const vegetable = vegetables.splice(vegetables.findIndex(veggie => veggie.id === id), 1)
+    res.status(200).json(vegetable)
+  } else {
+    // if the vegetable is not found, return error message and status code
+    const message = `Could not find vegetable with ID of ${id}`
+    next({ status: 404, message })
+  }
+})
+
+// update existing vegetable
+app.put('/vegetables/:id', helpers.validate, (req, res, next) => {
+  const { vegetables } = data
+  const { id } = req.params
+
+  // if the ID doesn't match an existing vegetable, give error
+  if ( !id ) {
```

Comment:
You'll actually need to check to make sure the ID matches one of the existing vegetables. Right now, if I put in an ID of `1`, if will still get into your `else` case.